src/api/gpu/ucx/nixl_device.cuh

```diff
@@ -30,52 +34,102 @@ struct nixlGpuSignal {
  * @brief An enumeration of different levels for GPU transfer requests.
  */
 enum class nixl_gpu_level_t : uint64_t {
-    THREAD,
-    WARP,
-    BLOCK,
-    GRID
+    THREAD = UCP_DEVICE_LEVEL_THREAD,
+    WARP   = UCP_DEVICE_LEVEL_WARP,
+    BLOCK  = UCP_DEVICE_LEVEL_BLOCK,
+    GRID   = UCP_DEVICE_LEVEL_GRID
+};
+
+/**
+ * @brief Parameters for GPU transfer requests with safe type conversion.
+ */
+struct nixlGpuXferReqParams {
+    nixlGpuXferReqParams() = delete;
+    __device__ nixlGpuXferReqParams(nixlGpuXferReqH req_hndl,
+                                    bool is_no_delay,
+                                    nixlGpuXferStatusH *xfer_status) :
+        mem_list{static_cast<ucp_device_mem_list_handle_h>(req_hndl)},
+        flags{is_no_delay ? static_cast<uint64_t>(UCP_DEVICE_FLAG_NODELAY) : 0},
+        ucp_request{xfer_status ? &xfer_status->device_request : nullptr} {}
```

Comment:
can be in next PR, but IMO to avoid this line we can define nixlGpuXferStatusH to be ucp_device_request_t* in nixl_device.cuh